lms/tasks/email_digests.py

```diff
@@ -66,9 +72,27 @@ def send_instructor_email_digest_tasks(*, batch_size):
                 )
             ).cte("candidate_courses")
 
-            h_userids = request.db.scalars(
-                select(User.h_userid)
-                .select_from(User)
+            tasks_done_subsubquery = (
+                select(
+                    TaskDone.data,
+                    func.rank()  # pylint:disable=not-callable
+                    .over(
+                        order_by=TaskDone.data["created_before"].desc(),
+                        partition_by=TaskDone.data["h_userid"],
+                    )
+                    .label("rank"),
+                ).where(TaskDone.data["h_userid"].as_string() == User.h_userid)
```

Comment:
FIXME: Probably want to add a couple more clauses to this `where()`. Maybe only consider `task_done`'s that have `"type": "instructor_email_digest"` in the `data` column, and maybe also only ones that have `"created_before"` in the `data` column. We probably can't go so far as to make sure that the `created_before` is a valid date string.